engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1453,6 +1552,42 @@ namespace dmGameSystem
         return 0;
     }
 
+    static int Physics_SetListener(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+        int top = lua_gettop(L);
+        int functionref = 0;
+        if (top >= 1) // completed cb
+        {
+            if (lua_isfunction(L, 1))
+            {
+                lua_pushvalue(L, 1);
+                // NOTE: By convention m_FunctionRef is offset by LUA_NOREF, in order to have 0 for "no function"
+                functionref = dmScript::RefInInstance(L) - LUA_NOREF;
+            }
+        }
+        dmScript::GetGlobal(L, PHYSICS_CONTEXT_HASH);
+        PhysicsScriptContext* context = (PhysicsScriptContext*)lua_touserdata(L, -1);
+        lua_pop(L, 1);
+
+        dmGameObject::HInstance sender_instance = CheckGoInstance(L);
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(sender_instance);
+        void* world = dmGameObject::GetWorld(collection, context->m_ComponentIndex);
+        if (world == 0x0)
+        {
+            return DM_LUA_ERROR("Physics world doesn't exist. Make sure you have at least one physics component in collection.");
+        }
```

Comment:
I think it's probably better to check this before we grab a reference to the function callback?